firmware/thruna/FSM/Src/Io/Io_AcceleratorPedals.c

```diff
@@ -1,76 +1,127 @@
     const float pot_len_mm    = PAPPS_RAW_VOLTAGE_TO_LEN_MM(pedal_voltage);
 
-    // Calculate pedal percentage
-    float pedal_percentage = (pot_len_mm - FULLY_UNPRESSED_POT_LENGTH_MM) /
-                                 (FULLY_PRESSED_POT_LENGTH_MM - FULLY_UNPRESSED_POT_LENGTH_MM) * 100.0f -
-                             PAPPS_DEADZONE_PERCENT;
+    float pedal_angle = Io_AcceleratorPedals_Angle_calc(
+        PAPPS_COS_LAW_COEFFICIENT, PAPPS_COS_LAW_DENOMINATOR, pot_len_mm, PAPPS_LENGTH_UNPRESSED_MM);
+    const float max_angle = Io_AcceleratorPedals_Angle_calc(
+        PAPPS_COS_LAW_COEFFICIENT, PAPPS_COS_LAW_DENOMINATOR, PAPPS_LENGTH_FULL_PRESSED_MM, PAPPS_LENGTH_UNPRESSED_MM);
 
-    return CLAMP(pedal_percentage, 0.0f, 100.0f);
-}
+    float pedal_percentage_raw = (pedal_angle / max_angle) * 100.0f;
+
+    if(pedal_percentage_raw <= DEAD_ZONE_PERCENT){
+        return 0;
+    }
+
+    float pedal_percentage = CLAMP(((100 / (100 - DEAD_ZONE_PERCENT)) * (pedal_percentage_raw - DEAD_ZONE_PERCENT)), 0.0f, 100.0f);
+    
+    return pedal_percentage;
+ }
 
-bool Io_AcceleratorPedals_PappsOCSC(void)
+
+bool Io_AcceleratorPedals_PappsOCSC(
+    void) // returns false if within range (no OC or SC) and true if it is outside the range (OS or SC is present)
 {
-    float pedal_voltage = 0.5f;
-    return !(MIN_PEDAL_VOLTAGE <= pedal_voltage && pedal_voltage <= MAX_PEDAL_VOLTAGE);
+    float pedal_voltage = Io_Adc_GetChannelVoltage(ADC1_CHANNEL_2);
+    return !(PAPPS_MIN_V <= pedal_voltage && pedal_voltage <= PAPPS_MAX_V);
 }
 
+
 float Io_AcceleratorPedals_GetSapps(void)
 {
     // TODO implement IO functionality
     // Length calculated via voltage
     float       pedal_voltage = Io_Adc_GetChannelVoltage(ADC1_CHANNEL_1);
-    const float pot_len       = PAPPS_RAW_VOLTAGE_TO_LEN_MM(pedal_voltage);
+    const float pot_len_mm    = PAPPS_RAW_VOLTAGE_TO_LEN_MM(pedal_voltage);
 
-    // Compute the angle relative to the y-axis with cosine law
-    const float pedal_travel_angle =
-        ((float)M_PI_2 -
-         (-(PAPPS_COS_LAW_COEFFICIENT - (pot_len * pot_len / PAPPS_COS_LAW_DENOMINATOR)) + (float)M_PI_2));
+    float pedal_angle = Io_AcceleratorPedals_Angle_calc(
+        SAPPS_COS_LAW_COEFFICIENT, SAPPS_COS_LAW_DENOMINATOR, pot_len_mm, SAPPS_LENGTH_UNPRESSED_MM);
+    const float max_angle = Io_AcceleratorPedals_Angle_calc(
+        SAPPS_COS_LAW_COEFFICIENT, SAPPS_COS_LAW_DENOMINATOR, SAPPS_LENGTH_FULL_PRESSED_MM, SAPPS_LENGTH_UNPRESSED_MM);
 
-    float secondary_angle = pedal_travel_angle * 180 / (float)M_PI + 16.3f;
-    return secondary_angle / 30.0f * 100.0f - SAPPS_DEADZONE_PERCENT;
+    float pedal_percentage_raw = (pedal_angle / max_angle) * 100.0f;
+
+    if(pedal_percentage_raw <= DEAD_ZONE_PERCENT){
+        return 0;
+    }
+
+    float pedal_percentage = CLAMP(((100 / (100 - DEAD_ZONE_PERCENT)) * (pedal_percentage_raw - DEAD_ZONE_PERCENT)), 0.0f, 100.0f);
```

Comment:
optional suggestion: This logic could be moved to another help function `applyDeadzone` or something like that